docs/api/security/Devicekey.md

```diff
@@ -19,31 +19,62 @@ The characteristics required by this root of trust are:
 
 The DeviceKey feature keeps the root of trust key in internal storage, using the KVStore component. Internal storage provides protection from external physical attacks to the device.
 
-The root of trust is generated at the first use of DeviceKey if the true random number generator is available in the device. If no true random number generator is available, you must pass the injected root of trust key to the DeviceKey before you call the key derivation API.
+The root of trust must be created before its first use. Otherwise, the key derivation API fails.
 
 ## Key derivation API
 
 `generate_derived_key`: This API generates a new key based on an array of data ([salt](https://en.wikipedia.org/wiki/Salt_(cryptography)) the caller provides. A single salt value always generates the same key, so if you need a new key, you must use a new salt value. The salt can have any value - array, string and so on.
 
-The generated keys can be 128 or 256 bits in length.
+The generated keys can be 128b or 256b in length.
 
-### Root of Trust Injection API
+### Root of Trust generation API
 
-`device_inject_root_of_trust`: You must call this API once in the lifecycle of the device, before any call to key derivation, if the device does not support true random number generator (`DEVICE_TRNG` is not defined).
+DeviceKey class needs root of trust ready to use before the derivation API's first call. There are two options to achieve this:
+
+- Create a device key using a built-in random number generator.
+- Manually fill the device key data array.
+
+Both cases requires injecting this key data to the KVStore reserved area.
+
+When `DEVICE_TRNG` is defined, the device supports a random number generator, and you may generate the key by calling `generate_root_of_trust()`. The call succeeds only if the key does not already exist. Subsequent calls will fail if you don't modify the existing key.
```

Comment:
> Subsequent calls will fail if you don't modify the existing key.

Sorry, I was probably unclear here. There is no way to modify the key, this sentence needs rewrite.

I tried to impress that you can only generate the key once. Modifying it, or regenerating it is not possible.

So subsequent call to this API will fail. And those calls don't modify the key.

Would it be clearer with:
```
The call succeeds only if the key does not already exist. Existing key cannot be changed.
```